protos/flyteidl/core/execution.proto

```diff
@@ -34,6 +35,7 @@ message NodeExecution {
         SKIPPED = 7;
         TIMED_OUT = 8;
         DYNAMIC_RUNNING = 9;
+        RECOVERED = 10;
```

Comment:
we should also update the statemachine diagram to indicate this new state in the docs. Also while you are at it, you can use the fancy sphinx plugin that Yee introduced

Reply:
good point, but i will do that in separate PR since it's in the flyte repo

Reply:
sure, this is to indicate that although a node succeeded it wasn't really run, rather it was recovered. if you think this is unnecessary i can just default to success

Reply:
aha.. so this is a terminal state that a node will move into when it's been recovered... cool cool.. SG!